torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -93,6 +93,28 @@ at::Tensor DoBinaryOp(const at::Tensor& self, const at::Tensor& other,
   return bridge::AtenFromXlaTensor(result);
 }
 
+template <typename B>
+at::Tensor DoBinaryOp(const at::Tensor& self, at::Scalar& other,
```

Comment:
No need to pass a writeable reference for the Scalar.
Use const&

Reply:
XlaTensor::add() and other scalar version of operations takes a non const Scalar

```
XLATensor XLATensor::add(const XLATensor& input, at::Scalar other,
                         at::Scalar alpha) {
```

because GetIrValueForScalar takes an non const Scalar
```
ir::Value XLATensor::GetIrValueForScalar(at::Scalar value,
                                         xla::PrimitiveType type,
                                         const Device& device) {
  if (IsSpecialScalar(value)) {
    return ir::ops::ScalarOp(std::move(value), type);
  }
  xla::ComputationClient::DataPtr data =
      GetDeviceData(value, TensorTypeFromXlaType(type), device);
  return ir::MakeNode<ir::ops::DeviceData>(std::move(data));
}
```
I think we might want to keep Scalar as a reference here, since `move` usually takes a non const object.

Reply:
In that case there is no difference between `Scalar&` and `const Scalar&`.
Never pass writeable references when not needed.
We also have a restriction in our rules to not use them. We have a few examples in tensor.h because the signatures of the ops we wanted to keep them similar to the PyTorch ones.
So, either pass `const Scalara&` or, given that `Scalar` is tiny, pass by value.

Reply:
Yeah, I overlooked that Scalar is pass by value to the XlaTensor functions. I will change all of the lambda function we passed to doBinaryOps 
```
-  return DoBinaryOp(self, other, [&](XLATensor& xself, at::Scalar& other) {
+  return DoBinaryOp(self, other, [&](const XLATensor& xself, const at::Scalar& other) {
```
to make it consistent. Will post it for review after test finish running.